trpcultivate_phenotypes/src/TripalCultivateValidator/TripalCultivatePhenotypesValidatorBase.php

```diff
@@ -201,4 +212,39 @@ public function getConfigAllowNew() {
 
     return $allownew;
   }
+
+  /**
+   * Split or explode a data file line/row values into an array using a delimiter.
+   * 
+   * @param string $row
+   *   A line in the data file.
+   * 
+   * @return array
+   *   An array containing the values extracted from the line after splitting it based
+   *   on a delimiter value.
+   */
+  public function splitRowIntoColumns(string $row) {
+    // Delimiter:
+    $delimiter = $this->context['delimiter'] ?? ''; 
+    if (empty($delimiter)) {
+      throw new \Exception(t('No delimiter provided.'));
+    }
+    
+    // Split the values.
+    $values = explode($delimiter, $row);
+
+    if (count($values) == 1 && $values[0] === $row) {
+      // The delimiter failed to split the row and returned the original row.
+      throw new \Exception(t('The data row or line provided could not be split using the delimiter.'));  
```

Comment:
Again no translation. 
Also you should include what delimiter this method used that failed as it can be helpful when debugging.

Reply:
Delimiter now part of the exception message 👍